src/components/shared/VirtualList.js

```diff
@@ -16,10 +16,11 @@ type VirtualListRowProps = {|
   +item: *,
   +index: number,
   +columnIndex: number,
+  // These properties are not used directly, but are needed for strict equality
+  // checks so that the components update correctly.
   +isSpecial: boolean,
-  // Items are not used directly, but are needed for strict equality checks so that
-  // the components update correctly.
   +items: *,
+  +forceRender?: number,
```

Comment:
I think this type definition needs more comments to explain what's going on.

Reply:
yeah, I think this whole file needs more comments. I can try to add some more as part of this PR.

Reply:
```js
  // These properties are not used directly, but are needed for strict equality
  // checks so that the components update correctly.
  // * `isSpecial` is used when we want to update one row or a few rows only,
  //   this is typically when the selection changes and both the old and the new
  //   selection need to be changed.
  +isSpecial: boolean,
  // * `items` contains the full items, so that we update the whole list
  //   whenever the source changes. This is necessary because often `item` is a
  //   native value (eg a number), and shallow checking only `item` won't always
  //   give the expected behavior.
  +items: *,
  // * `forceRender` is passed through directly from the main VirtualList
  //   component to the row as a way to update the full list for reasons
  //   unbeknownst to this component. This can be used for example in chart-like
  //   panels where we'd want to redraw if some source value necessary to the
  //   computation changes.
  +forceRender?: number | string,
```

I'll also add an introduction at the top of the file.